python/ray/services.py

```diff
@@ -493,45 +493,32 @@ def start_redis(node_ip_address,
 
     if use_credis is None:
         use_credis = ("RAY_USE_NEW_GCS" in os.environ)
-    if use_credis and password is not None:
-        # TODO(pschafhalter) remove this once credis supports
-        # authenticating Redis ports
-        raise Exception("Setting the `redis_password` argument is not "
-                        "supported in credis. To run Ray with "
-                        "password-protected Redis ports, ensure that "
-                        "the environment variable `RAY_USE_NEW_GCS=off`.")
-    if not use_credis:
-        assigned_port, p = _start_redis_instance(
-            node_ip_address=node_ip_address,
-            port=port,
-            redis_max_clients=redis_max_clients,
-            stdout_file=redis_stdout_file,
-            stderr_file=redis_stderr_file,
-            password=password,
-            # Below we use None to indicate no limit on the memory of the
-            # primary Redis shard.
-            redis_max_memory=None)
-        processes.append(p)
-    else:
-        assigned_port, p = _start_redis_instance(
-            node_ip_address=node_ip_address,
-            port=port,
-            redis_max_clients=redis_max_clients,
-            stdout_file=redis_stdout_file,
-            stderr_file=redis_stderr_file,
-            executable=CREDIS_EXECUTABLE,
-            # It is important to load the credis module BEFORE the ray module,
-            # as the latter contains an extern declaration that the former
-            # supplies.
-            modules=[CREDIS_MASTER_MODULE, REDIS_MODULE],
-            password=password,
-            # Below we use None to indicate no limit on the memory of the
-            # primary Redis shard.
-            redis_max_memory=None)
-        processes.append(p)
-    if port is not None:
-        assert assigned_port == port
-    port = assigned_port
+    if use_credis:
+        if password is not None:
+            # TODO(pschafhalter) remove this once credis supports
+            # authenticating Redis ports
+            raise Exception("Setting the `redis_password` argument is not "
+                            "supported in credis. To run Ray with "
+                            "password-protected Redis ports, ensure that "
+                            "the environment variable `RAY_USE_NEW_GCS=off`.")
+        assert num_redis_shards == 1, \
```

Comment:
Instead of using `\` use something like

```python
assert num_redis_shards == 1, (
    "For now, RAY_USE_NEW_GCS supports 1 shard, and credis "
    "supports 1-node chain for that shard only.")
```